src/commands/agent/generate/template.ts

```diff
@@ -31,32 +62,132 @@ export default class AgentGenerateTemplate extends SfCommand<AgentGenerateTempla
       summary: messages.getMessage('flags.agent-api-name.summary'),
       required: true,
     }),
-    'output-dir': Flags.directory({
-      char: 'd',
-      exists: true,
-      summary: messages.getMessage('flags.output-dir.summary'),
+    'agent-version': Flags.integer({
+      summary: messages.getMessage('flags.agent-version.summary'),
+      required: true,
     }),
   };
 
   public async run(): Promise<AgentGenerateTemplateResult> {
     const { flags } = await this.parse(AgentGenerateTemplate);
+    const { 'agent-api-name': botName, 'agent-version': botVersion } = flags;
+
+    const parser = new XMLParser({ ignoreAttributes: false });
+    const builder = new XMLBuilder({ format: true, ignoreAttributes: false, indentBy: '    ' });
 
-    // TODO: look for a Bot with the agent API name
-    const botName = flags['agent-api-name'];
-    const outputDir = flags['output-dir'] ? resolve(flags['output-dir']) : this.project?.getDefaultPackage().fullPath;
+    // Since we are cloning the GenAiPlanner, we need to use a different name than the Agent (Bot) we started with
+    // We will use this name for the BotTemplate also to make it clear they are related
+    const finalFilename = `${botName}_v${botVersion}_Template`;
 
-    const agentTemplateFilePath = join(outputDir as string, 'agentTemplates', `${botName}.agentTemplate-meta.xml`);
-    mkdirSync(dirname(agentTemplateFilePath), { recursive: true });
+    // TODO: Should we add a flag to specify the packagePath?
+    const packagePath = this.project?.getDefaultPackage().fullPath;
+    const basePath = join(packagePath as string, 'main', 'default');
```

Comment:
I don't think we can make this assumption ("main/default" in the default package dir of the project).  What about these options?
1. Use SDR to find source components in the project by the metadata name (api name) and then use those paths as the base.
OR
2. Rather than have an agent-api-name flag, ask for a path to the Bot file that should be used for the scaffolding.  Then everything is relative to that path.

Reply:
@iowillhoit - I'm OK with suggestion # 2.  @shetzel makes a good point about being more explicit.

Could we validate that the file specified by the path is a `*.bot-meta.xml` file?

Reply:
Yea I can add a simple check for that. The command would fail pretty quickly for other reasons if a Bot file was not passed